port/unix/omrsl.c

```diff
@@ -438,3 +439,79 @@ omrsl_startup(struct OMRPortLibrary *portLibrary)
 {
 	return 0;
 }
+
+/**
+ * Native Libraries
+ *
+ * This function is called go get all the shared libraries loaded by a process.
+ *
+ * @param[in] portLibrary Pointer to the OMR port library.
+ * @param[in] callback Function to be called for each library.
+ * @param[in] userData User-defined data passed to the callback.
+ *
+ * @return 0 if successful, or the first non-zero return value from the callback.
+ */
+uintptr_t
+omrsl_get_libraries(struct OMRPortLibrary *portLibrary, OMRLibraryInfoCallback callback, void *userData)
+{
+#if defined(LINUX)
+/* Length of buffer. */
+#define READ_CHUNK_SIZE (PATH_MAX + 100)
+	char buffer[READ_CHUNK_SIZE + 1];
+	uintptr_t result = 0;
+	void *addrLow = NULL;
+	void *addrHigh = NULL;
+	unsigned long offset = 0;
+	int devMajor = 0;
+	int devMinor = 0;
+	unsigned long inode = 0;
+	char permissions[5] = {0};
+	int count = 0;
+	int path_offset = 0;
+	int32_t portableError = 0;
+	intptr_t fd = portLibrary->file_open(portLibrary, "/proc/self/maps", EsOpenRead, 0);
+	if (-1 == fd) {
+		portableError = portLibrary->error_last_error_number(portLibrary);
+		Trc_PRT_failed_to_open_proc_maps(portableError);
+		portLibrary->error_set_last_error_with_message(
+				portLibrary,
+				portableError,
+				"Failed to open /proc/self/maps");
+		return (uintptr_t)(intptr_t)portableError;
+	}
+	while (NULL != portLibrary->file_read_text(portLibrary, fd, buffer, sizeof(buffer))) {
+		size_t len = strlen(buffer);
+		if ((len > 0) && (buffer[len-1] != '\n')) {
+				portableError = portLibrary->error_last_error_number(portLibrary);
+				portLibrary->error_set_last_error_with_message(
+						portLibrary,
+						portableError,
+						"An unexpected long path name. Line is truncated.");
+				continue;
+		}
```

Comment:
Suggest using `strchr()` here:
```c
		if (NULL == strchr(buffer, '\n')) {
```
Please correct the indentation of lines 485-490.